docs/design/auto-storage-scale.md

```diff
@@ -0,0 +1,49 @@
+# Automatic Storage Scaling
+
+# Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+# Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+# High-Level Design
+
+- Create a AutoStorageScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: AutoStorageScaling
+    Metadata:
+	    Name: <storageclusterName>-<deviceClassName>
```

Comment:
The name is not important, right? It can be anything. The spec fields would have the important settings.

Reply:
@travisn this is a suggestion I left in the Design doc just to identify at-a-glance the storagecluster and deviceclass which this CR is associated to.

Reply:
Correct name is not important, it has to be unique

Reply:
Sounds good as long as it's just a convention, and we aren't using that to extract details